docs/source/tutorials.md

```diff
@@ -101,3 +101,197 @@ Since the target repository is different than the source, it has to be specified
 ```bash
 agda2hs ./src/agda/Usage.agda -o ./src/haskell/
 ```
+## How (and what) to prove?
+
+[Example source code](https://github.com/agda/agda2hs/tree/master/tutorials/example-proofs)
+
+This tutorial aims to explain how to apply different formal verification techniques compliant with agda2hs and haskell. 
+
+### Constructor constraints
+
+The code described in this section can be found in the file `Triangle.agda`.
+
+Lets say we want to have a data type describing a triangle. A first attempt might look somewhat like this:
+
+```agda
+open import Haskell.Prelude
+
+data Triangle : Set where
+    MkTriangle : (alpha beta gamma : Nat)
+           → Triangle
+
+{-# COMPILE AGDA2HS Triangle #-}
+```
+It's defined with three angles, because maybe thats the only property we are interested in so far. However, three arbitrary angle values do not make a triangle. First of all, they cannot be negative - to make things easier we use natural numbers (`Nat`) to represent the angle values. But that is not enough, three angles can constitute a triangle only if they sum up to 180 degrees and at most one angle is right or obtuse. This can be modeled in Agda:
```

Comment:
Isn't it enough to have all three angles be strictly larger than 0 and sum up to 180?

Reply:
This depends on if we want to include degenerate triangles in our definition of valid triangles, as they are the only edge case difference between these two given definitions. Should I elaborate about this in the tutorial? I thought it might be a bit off-topic to go into great details of geometry definitions.

Reply:
Probably not worth going into any details here, but perhaps you could consider just using the simple definition.